app/controllers/ViewArrivalNotificationsController.scala

```diff
@@ -17,35 +17,53 @@
 import play.api.i18n.I18nSupport
 import play.api.libs.json.Json
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import renderer.Renderer
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import viewModels.{ViewArrivalMovements, ViewMovement}
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
-class ViewArrivalNotificationsController @Inject()(renderer: Renderer,
-                                                   val controllerComponents: MessagesControllerComponents,
-                                                   appConfig: FrontendAppConfig,
-                                                   destinationConnector: DestinationConnector)
-                                                  (implicit ec: ExecutionContext)
-  extends FrontendBaseController with I18nSupport {
+class ViewArrivalNotificationsController @Inject()(
+                                                    renderer: Renderer,
+                                                    val controllerComponents: MessagesControllerComponents,
+                                                    destinationConnector: DestinationConnector,
+                                                    referenceDataConnector: ReferenceDataConnector
+                                                  )(implicit ec: ExecutionContext, appConfig: FrontendAppConfig)
+  extends FrontendBaseController
+    with I18nSupport {
 
-  def onPageLoad: Action[AnyContent] = Action.async {
-    implicit request =>
+  def onPageLoad: Action[AnyContent] = Action.async { implicit request =>
+    destinationConnector.getMovements().flatMap {
+      movements =>
+        val urls = Json.obj(
+          "declareArrivalNotificationUrl" -> appConfig.declareArrivalNotificationUrl,
+          "homePageUrl" -> routes.IndexController.onPageLoad().url)
 
-      destinationConnector.getArrivalMovements.flatMap {
-        dataRows =>
-
-          val json = Json.obj(
-            "declareArrivalNotificationUrl" -> appConfig.declareArrivalNotificationUrl,
-            "dataRows" -> dataRows
-          )
+        Future.sequence(movements.map(convertToViewMovements)).map(ViewArrivalMovements.apply)
+          .map(Json.toJsObject[ViewArrivalMovements])
+          .flatMap(json =>
+            renderer.render("viewArrivalNotifications.njk", json ++ urls).map(Ok(_)))
+    }
+  }
 
-          renderer.render("viewArrivalNotifications.njk", json).map(Ok(_))
-      }
+  private def convertToViewMovements(movement: Movement)(implicit hc: HeaderCarrier): Future[ViewMovement] = {
+    referenceDataConnector.getCustomsOffice(movement.presentationOfficeId) map {
+      presentationOffice =>
+        ViewMovement(movement.date,
+          movement.time,
+          movement.movementReferenceNumber,
+          movement.traderName,
+          movement.presentationOfficeId,
+          presentationOffice.name,
```

Comment:
Should we investigate having a fallback in the event that the API cannot return the name for Presentation office?